### ISSUE TITLE
4.3.0 Android - clicks on pins no longer fires a mapclick event

### ISSUE THREAD

**Issue by mas222** [tidev/ti.map#231]
clicks on pins no longer fires a mapclick event when the map is set up for clustering, it will however show the info box above it and if that is clicked the correct mapclick event will be fired.

**Comment by m1ga** [comment]
Tried to solve this today but couldn't find a solution to cover all cases. The problem is 
`map.setOnMarkerClickListener(this);`
or
```
map.setOnMarkerClickListener(mClusterManager);
mClusterManager.setOnClusterClickListener(this);
mClusterManager.setOnClusterItemClickListener(this);
```

The first one will handle the normal marker clicks. The second will use `onClusterClick` (clusters -> zoom to break up cluster) and `onClusterItemClick` (click on unclustered items).
You can add the pin callback to the `onClusterItemClick` which would work. But when you mix clustered and non clustered items you can't listend to both `setOnMarkerClickListener` as far as I see.

One option would be to decide if you use clustering and then use the second one and if not use the first one. That would work again if you don't mix clustered and non-clustered items

**Comment by andrzejschmidt** [comment]
Hey @mas222, have you found any workaround for that by any chance? I have an exact same problem trying to show custom annotation on clicking the pin (worked until 4.2.0), but also want to take advantage of new clustering feature (from 4.3.0).
Tried m1ga's suggestions but I end up with errors trying to build the module.

**Comment by ypbnv** [comment]
@andrzejschmidt I created a ticket for the issue. You can track there what is the status.

**Comment by zainesmail** [comment]
Is there going to be a  new release out soon for 4.3?

**Comment by AppsolutEinfach** [comment]
Seems to be fixed in [4.3.1](https://github.com/appcelerator-modules/ti.map/releases/tag/android-4.3.1).

**Comment by hansemannn** [comment]
@garymathews can you close this one?
